### ISSUE TITLE
[Kev 5] Pointer head + packed forward conformance

### ISSUE THREAD

**Issue by AtlantisPleb** [OpenAgentsInc/openagents#9342]
Part of #9337.

## What to pull over

From `kev/model.py:PointerHead` and `DecisionModel.forward`:

- Two linear maps hidden→256 (query from the `<decide>` hidden state, key from each `</opt>` hidden state), scaled dot product `1/sqrt(256)`, softmax per question.
- `head.safetensors` loading (converted in Kev-1).
- `forward(record) → [logits_k] per question → softmax`; the packed-vs-separate path used for verification.

## Acceptance

- For every corpus fixture, Rust per-question probabilities match the golden outputs within fp32 tolerance on CPU (target max delta ≤ 1e-4; Python packed-vs-separate delta is 3.7e-6).
- Packed and separate Rust evaluations agree within the same bound.
- Mechanism fixtures reproduced: isolation secret stays at chance in sibling, reads ~1.0 in state.

Generated with [Devin](https://devin.ai)

**Comment by AtlantisPleb** [comment]
Done in 2d8f237291.

- `crates/kev/src/head.rs` — `PointerHead`: q/k linear maps 896→256, scaled dot product `1/sqrt(256)`, softmax per question. Loads `head.safetensors` from the artifact bundle.
- `crates/kev/src/decision.rs` — `DecisionModel`: backbone + LoRA + head + tokenizer assembled from the artifact dirs; `encode`/`probs`/`systemone` serving path.
- `tests/conformance.rs`: all golden probabilities within tolerance (max delta 4.1e-6) and every rounded answer byte-identical; packed-vs-separate delta 1.2e-6 (reference fixture bound 2.21e-6); isolation probe 0.056/0.056/0.997 matching the reference; forgery option count unchanged.

**Comment by AtlantisPleb** [comment]
Cross-lane evidence from the Lev comparison harness (`lev-compare`, #9354), which sends the same items through one unmodified `crates/jev` client to every door:

| Item | Truth | jev (hosted) | kev-serve | lev-serve |
| --- | --- | --- | --- | --- |
| easy/charge | billing | billing | **other** | billing |
| easy/crash | technical | technical | **other** | technical |
| easy/quote | sales | sales | **other** | sales |
| easy/offtopic | other | other | other | other |
| hard/upgrade | billing | billing | **other** | technical |
| hard/seats | billing | billing | **other** | billing |
| hard/slow | technical | technical | **other** | technical |
| hard/renewal | sales | sales | **other** | sales |

`kev-serve` scored 1 of 8, answering `other` on every item except the one whose answer happens to be `other`, at a mean confidence of 0.81 with 4 confident errors. A constant argmax rather than a wrong-but-varying one looks like a readout that is not reading — an option-position or gather index, or the `</opt>` and `<decide>` hidden states, rather than a quality gap. `kev-0.5b`'s own card reports 0.94 on a four-option news classification, so this is a long way from the checkpoint's measured behavior.

Run against `~/work/kev-artifacts/kev-0.5b` + `qwen25-0.5b` at commit 0fc8a95d90 on CPU. Reproduce with:

```
./target/release/kev-serve --adapter-dir ~/work/kev-artifacts/kev-0.5b --base-dir ~/work/kev-artifacts/qwen25-0.5b --port 8009
cargo run -p lev --features serve --bin lev-compare -- --door kev=http://127.0.0.1:8009
```

Posting here rather than fixing it, since the port is yours and the golden-probability conformance in this issue is exactly the check that should catch it. Not touching `crates/kev`.

**Comment by AtlantisPleb** [comment]
**Correction — retracting my earlier comment. `crates/kev` is not broken, and my diagnosis was wrong.**

Two checks I should have run before posting:

1. `cargo test --release -p kev --test conformance` passes all five, including `golden_probabilities_reproduce`.
2. Posting the golden `support.json` request to a live `kev-serve` reproduces the fixture **exactly** — `returns` at 0.95/0.03/0.02, conf 0.93; noul 0.75; score 0.99 with 0.18/0.66/0.17. The serve path matches the test path.

The 1 of 8 was my harness, not your port. My Choice included an `other` option, and `kev-0.5b` collapses onto it. That is a documented property of the checkpoint rather than a defect: its card records none-of-the-above augmentation at p=0.10, and the research log in `docs/kev/model-cards.md` names the fix that came later — "None-of-the-above minimal pairs fixed the 'none' shortcut in-domain (0.75 → 0.93 at 4B)". `kev-0.5b` predates it.

Same seven items with the `other` option removed, same door:

| Item | Truth | kev-0.5b | Confidence |
| --- | --- | --- | --- |
| easy/charge | billing | billing | 0.99 |
| easy/crash | technical | technical | 1.00 |
| easy/quote | sales | sales | 0.97 |
| hard/upgrade | billing | **technical** | 0.58 |
| hard/seats | billing | billing | 0.61 |
| hard/slow | technical | technical | 1.00 |
| hard/renewal | sales | sales | 0.96 |

6 of 7, and the one miss is the deliberately ambiguous item that hosted Jev also had to reason about and that Lev misses too — at 0.58 confidence, which is the model correctly reporting that it is unsure.

Apologies for the noise. The none-shortcut is worth knowing for #9356's multi-variant work, since it is one of the things the larger bases are reported to fix, and I will carry both the with-`other` and without-`other` variants in the comparison harness so the shortcut shows up as a measured property instead of looking like a bug.
